presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -125,24 +128,24 @@ public BaseJdbcClient(BaseJdbcConfig config, String identifierQuote, ConnectionF
         this(
                 identifierQuote,
                 connectionFactory,
+                config.getJdbcTypesMappedToVarchar(),
                 requireNonNull(config, "config is null").isCaseInsensitiveNameMatching(),
-                config.getCaseInsensitiveNameMatchingCacheTtl(),
-                config.getJdbcTypesMappedToVarchar());
+                config.getCaseInsensitiveNameMatchingCacheTtl());
     }
 
     public BaseJdbcClient(
             String identifierQuote,
             ConnectionFactory connectionFactory,
+            Set<String> jdbcTypesMappedToVarchar,
             boolean caseInsensitiveNameMatching,
-            Duration caseInsensitiveNameMatchingCacheTtl,
-            Set<String> jdbcTypesMappedToVarchar)
+            Duration caseInsensitiveNameMatchingCacheTtl)
     {
         this.identifierQuote = requireNonNull(identifierQuote, "identifierQuote is null");
         this.connectionFactory = requireNonNull(connectionFactory, "connectionFactory is null");
         requireNonNull(caseInsensitiveNameMatchingCacheTtl, "caseInsensitiveNameMatchingCacheTtl is null");
 
         this.caseInsensitiveNameMatching = caseInsensitiveNameMatching;
-        this.jdbcTypesMappedToVarchar = jdbcTypesMappedToVarchar;
+        this.jdbcTypesMappedToVarchar = ImmutableSet.copyOf(requireNonNull(jdbcTypesMappedToVarchar, "jdbcTypesMappedToVarchar is null"));
```

Comment:
Please move this up, after `this.connectionFactory =`

Reply:
Sorry, missed that one ...

Reply:
Fixed